portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,15 +30,37 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ArrayList<String> greetings = new ArrayList<String>();
-    greetings.add("Hello Tayyaba!");
-    greetings.add("Hello User!");
-    greetings.add("Hello World!");
     String json = convertToJsonUsingGson(greetings);
     response.setContentType("text/html;");
     response.getWriter().println(json);
   }
 
-  private Gson gson;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = getParameter(request, "text-input", "");
+
+    // Break the text into individual words.
+    String[] words = text.split("\\s*,\\s*");
```

Comment:
From this stackoverflow search[1], someone also suggests using

String[] words = s.split("\\W+");

as a more general solution for ASCII. Yours works as fine for words split by comma or space, so it's your call.

[1] https://stackoverflow.com/questions/11726023/split-string-into-individual-words-java